lib/generator/code_generator.dart

```diff
@@ -10,83 +10,112 @@ import 'package:graphql_generator/generator/objects.dart';
+        enumObjectTypes,
+        interfaceObjectTypes,
+        unionObjectTypes,
+        objectObjectTypes,
+        inputObjectTypes,
+        mutationObjectType);
+  }
+
+  List<ObjectType> getTypesFromList(
+      List<ObjectType> responseTypes, Kind typeKind) {
+    List<ObjectType> typeList =
+        responseTypes.where((type) => type.kind == typeKind).toList();
+    return typeList;
   }
 
-  splitTypes() {
-    enumTypes = responseTypes.where((type) => type.kind == Kind.ENUM).toList();
-    interfaceTypes =
-        responseTypes.where((type) => type.kind == Kind.INTERFACE).toList();
-    unionTypes =
-        responseTypes.where((type) => type.kind == Kind.UNION).toList();
-    inputObjectTypes =
-        responseTypes.where((type) => type.kind == Kind.INPUT_OBJECT).toList();
-    objectTypes =
-        responseTypes.where((type) => type.kind == Kind.OBJECT).toList();
-    mutation = responseTypes
+  ObjectType getMutationObjectType(
+      List<ObjectType> responseTypes, String mutationClassName) {
+    ObjectType mutationObjectType = responseTypes
         .where((type) => type.name == mutationClassName)
         .toList()
         .first;
+    return mutationObjectType;
   }
 
-  generateTypes() {
+  generateTypes(
+      String namespace,
+      Map<String, String> fragments,
+      Map<String, DartType> types,
+      List<ObjectType> enumObjectTypes,
+      List<ObjectType> interfaceObjectTypes,
+      List<ObjectType> unionObjectTypes,
+      List<ObjectType> objectObjectTypes,
+      List<ObjectType> inputObjectTypes,
+      ObjectType mutationObjectType) {
     var result = '';
     var emitter = DartEmitter(Allocator());
+    Map<String, Class> classes = {};
+
     List<String> enumString =
-    EnumGenerator().enumGenerator(enumTypes, namespace);
-    Map<String, Class> interfaces =
-    InterfaceGenerator().interfaceGenerator(interfaceTypes, namespace);
+        EnumGenerator().enumGenerator(enumObjectTypes, namespace: namespace);
+    Map<String, Class> interfaces = InterfaceGenerator().generate(
+        types,
+        enumObjectTypes,
+        interfaceObjectTypes,
+        unionObjectTypes,
+        objectObjectTypes,
+        inputObjectTypes,
+        namespace: namespace);
     Map<String, Class> unions =
-    UnionGenerator().unionGenerator(unionTypes, namespace);
-    Map<String, Class> inputs = InputObjectGenerator()
-        .inputObjectGenerator(inputObjectTypes, namespace, enumTypes);
-    Map<String, Class> objects = ObjectClassGenerator()
-        .objectClassGenerator(objectTypes, namespace, enumTypes, unionTypes);
+        UnionGenerator().generate(unionObjectTypes, namespace: namespace);
+    Map<String, Class> inputs = InputObjectGenerator().generate(
+        types,
+        enumObjectTypes,
+        interfaceObjectTypes,
+        unionObjectTypes,
+        objectObjectTypes,
+        inputObjectTypes,
+        namespace: namespace);
+    Map<String, Class> objects = ObjectClassGenerator().generate(
```

Comment:
You can use the `var` keyword instead of `Map<String, Class>` thanks to type inference. Please read up on type inference in Dart.

Reply:
We could replace all the variables and parameter to `var` right? So wouldn't that make it difficult for developers to understand the type of the variable when developing?

Reply:
It wouldn't make it too difficult because the vscode editor would tell you the type of variable that is being returned when hovering over it.